src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioErrorReportingService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Extensions;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using VS = Microsoft.VisualStudio.Progression;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation
+{
+    internal class VisualStudioErrorReportingService : IErrorReportingService
+    {
+        private VisualStudioWorkspaceImpl workspace;
+        private static InfoBarButton EnableItem = new InfoBarButton(ServicesVSResources.Enable);
+        private static InfoBarButton EnableAndIgnoreItem = new InfoBarButton(ServicesVSResources.EnableAndIgnore);
+
+        public VisualStudioErrorReportingService(VisualStudioWorkspaceImpl workspace)
+        {
+            this.workspace = workspace;
+        }
+
+        public async void ShowErrorInfoForCodeFix(DocumentId documentId, string codefixName, Action OnEnableClicked, Action OnEnableAndIgnoreClicked)
+        {
```

Comment:
if infobar should appear active document, can ShowErrorInfoFOrCodeFixAsync figure that out itself rather than asking caller to give that info?